config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java

```diff
@@ -172,6 +172,10 @@ public void validateName(Map<String, EnvironmentVariableConfig> variableConfigMa
         String currentVariableName = name.toLowerCase();
         String parentDisplayName = validationContext.getParentDisplayName();
         CaseInsensitiveString parentName = getParentNameFrom(validationContext);
+        if(currentVariableName.startsWith(" ") || currentVariableName.endsWith(" ")){
```

Comment:
@zabil — I'd not merged this because this is a really bad test.

Ideally we'd want to test for all whitespace, not just a space character. Basically test whether `currentVariableName.trim().equals(currentVariableName)`